### DISCUSSION TITLE
Turn off ligatures

### DISCUSSION THREAD

**Discussion by sdt16** [neovide/neovide#934] [Q&A] [answered]
Hi,

Is there a way to turn off ligature rendering? I love everything else (especially the cursor animations), but I can't stand ligatures :/

Thanks!

**Comment by Kethku** [comment, marked as answer]
Interesting. My recommendation generally is just use a font that doesn't have ligatures. I haven't thought about what would be necessary to incorrectly render fonts... Let me think if there is a way to cleanly disable ligatures if necessary.

**Reply by sdt16** [reply]
Ah, didn't realize the font itself has input into if ligatures are used. I'll try that. Thanks!

**Reply by cspiegel** [reply]
It shouldn't be necessary to "incorrectly" render a font, as fonts with ligatures tend to implement them as OpenType features. For Fira Code, for example, disabling the "calt" feature will turn off ligatures.

I'm not sure the level of effort it'd take, though.  With Harfbuzz it's a matter of passing features to the `hb_shape_full()` function, but I've got no familiarity outside of that (and really very little with Harfbuzz; just enough to do a bit of damage while trying to disable ligatures in Qt).

**Comment by wookayin** [comment]
+1 There should be an option to disable ligature. Simply "using a font that lacks ligature" makes no sense -- note that Nerd fonts (for instance) with unicode and symbol supports have ligature. All other apps can turn off ligature despite the same font being used.

**Reply by fredizzimo** [reply]
We have this PR open https://github.com/neovide/neovide/pull/1920

Unfortunately, it has stalled a bit due to us maintainers not being able to properly decide how we want it configured.

**Reply by MaskRay** [reply]
Would be nice to have this feature. I primarily use Fantasque Sans Mono and specifically for neovide I have to install a no-ligature variant https://github.com/RedBearAK/FantasqueSansMNoLigNerdFont 

> Fantasque Sans Mono font variant with no ligatures, no loop K, large line height, and processed with the Nerd Font conversion script.

**Reply by fredizzimo** [reply]
According to this, you can turn off the ligatures for that font https://github.com/belluzj/fantasque-sans/issues/122#issuecomment-1647770542

In the Neovide `config.toml`
```
[font.features]
"Font Name Goes Here" = ["-calt"]
```

Also see this issue https://github.com/neovide/neovide/issues/1037

**Reply by austindelic** [reply]
> According to this, you can turn off the ligatures for that font [belluzj/fantasque-sans#122 (comment)](https://github.com/belluzj/fantasque-sans/issues/122#issuecomment-1647770542)
> 
> In the Neovide `config.toml`
> 
> ```
> [font.features]
> "Font Name Goes Here" = ["-calt"]
> ```
> 
> Also see this issue #1037

This worked for me!
